src/utils/helpers.ts

```diff
@@ -422,6 +429,24 @@ export const subIn256 = (x: bigint, y: bigint): bigint => {
   }
 };
 
+export const splitFee = (
+  amount: bigint,
+  settlerFeesInBps: bigint,
+  protocolShareBps: bigint
+): { amountIn: bigint; protocolShareAmount: bigint; senderShareAmount: bigint } => {
+  const feeAmount = (amount * settlerFeesInBps) / 10_000n;
+  const amountIn = amount - feeAmount;
```

Comment:
amountIn should remove senderFeeAmount as well. That's how it does in the protocol: https://github.com/meliopolis/chainhopper-protocol/blob/main/src/base/Settler.sol#L64.

Reply:
re: settlerFeesInBps. Total amount was right, it's only the split that's off due to that third variable.